### PR TITLE
perf: address rendering performance issues on trade input page

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

Reduces the amount of repeat work done in the trade input, namely within the related asset selectors and the rendering around the trade input components in general.

Two insights when debugging performance issue on trade input:
1. selectors using a parameter e.g `const xyz = useAppSelector(state => myselector(state, myParam))` will recompute on every access even if the selector was implemented with `createDeepEqualOutputSelector`. Instead, `createCachedSelector` is required.
2. `{ propA, propB, ...rest } = props` pattern for props causes re-render on every parent component render. We should avoid this and instead pass the rest as field in the props. Example is instead of `type MyProps = { propA: number } & BoxProps`, we should do `type MyProps = { propA: number; boxProps: BoxProps }`

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

NA

Low risk of broken related asset selectors.

> What protocols, transaction types or contract interactions might be affected by this PR?

## Testing

Check the correctness of related assets in the asset page, and in the chain drop-down in the trade input page. Pay particular attention to asset balances in the chain drop-down.

### Engineering

Note the use of `createCachedSelector`. The previous implementation used `createDeepEqualOutputSelector` which recomputes every access due to the presence of the `assetId` parameter.

### CODE REVIEW

**Comment by woodenfurniture** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/shapeshift/web/6272?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#6272** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6272?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6230** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6230?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6176** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6176?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6213** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6213?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6171** <a href="https://app.graphite.dev/github/pr/shapeshift/web/6171?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/6272?utm_source=stack-comment).

File: src/components/AssetSelection/components/AssetChainDropdown/AssetChainDropdown.tsx [hunk 1]
```diff
@@ -38,7 +42,6 @@ export const AssetChainDropdown: React.FC<AssetChainDropdownProps> = ({
   buttonProps,
   isLoading,
   isError,
-  ...menuProps
```
Comment (woodenfurniture) [inline src/components/AssetSelection/components/AssetChainDropdown/AssetChainDropdown.tsx:41]: spread operator causes re-render every frame

Comment (gomesalexandre) [reply src/components/AssetSelection/components/AssetChainDropdown/AssetChainDropdown.tsx:41]: Fine to remove in this case as we weren't using menu props, but indeed, const reassignment of `menuProps` every render is inherently dangerous, nice catch!

Don't think we should necessarily step away from that pattern, we could always `useMemo` when there's a legit use case for the rest props.

But I do agree it's always better to be fully declarative and pass all props explicitly, that would've avoided this guy here, where `<Menu />` doesn't actually doesn't consume any of said rest props and an effective empty object is passed

**Review by gomesalexandre — COMMENTED** [review]
Conceptual stamp on the perf. side of things here, pending runtime testing including https://github.com/shapeshift/web/pull/6230

**Review by gomesalexandre — APPROVED** [review]
Tested locally and didn't find any weirdnesses, though this will obviously warrant a full depth testing by ops:

- Unactionable quotes because of lack of gas are now visible 🎉 

https://github.com/shapeshift/web/assets/17035424/22adb4ac-fc3f-401f-9a80-d3cb3f579d77

- Rates look sane

image

- Percentage diff looks sane

image
image

- For paranoia's sake, trade confirm looks sane

image

- Debouncing looks sane

i.e 0.21 -> 0.2

image

While testing this, noticed one thing that's not necessarily a blocker but should definitely be tackled as a UX weirdness: the refetch countdown spinner is only present on the selected quote (i.e no selected quote, no spinner), despite it happening over all quotes, as can be seen on the recording below where all refetch as they should:


https://github.com/shapeshift/web/assets/17035424/bf432c2d-f21f-4453-9932-97941c5aa695

**Comment by woodenfurniture** [comment]
> While testing this, noticed one thing that's not necessarily a blocker but should definitely be tackled as a UX weirdness: the refetch countdown spinner is only present on the selected quote (i.e no selected quote, no spinner), despite it happening over all quotes, as can be seen on the recording below where all refetch as they should:

Correct yeah, this is part of the design change to minimise the crazy number of spinners and animations we had during fetching - the goal was to have 1 spinner for the form, and 1 for the selected quote:
image

**Pull request merged**